### ISSUE TITLE
grafana-server.service, code=exited, status=2/INVALIDARGUMENT  

### ISSUE THREAD

**Issue by MarkoBursic** [grafana/grafana#21139]
- Grafana version: 6.5
- Data source type & version: influxdb 1.7
- OS Grafana is installed on: Armbiant Buster (Debian) Rockchip RK3399
- User OS & Browser: Windows 10, Chrome

After two weeks of fine working, the Grafana won't start. It exits at line 117 of /grafana-server/main.go , which is the very begining of the start process. What could be the cause?

![mamilnik_2](https://user-images.githubusercontent.com/7830293/70990709-621e7c80-20c6-11ea-9b0f-5a9c96dcec32.jpg)

**Comment by aknuds1** [comment]
Can you find the originating error in the Grafana server logs?

**Comment by MarkoBursic** [comment]
You mean /var/log/grafana/grafana.log ? It has two lines with a date 5 days old, they both have msg="Request competed" and remote IP address information. All I have is an printscreen image of it, I can post it, but there is no useful information about the issue. Where could I find some more detailed log or how should I get some debug info?

**Comment by aknuds1** [comment]
How do you run Grafana exactly?

**Comment by MarkoBursic** [comment]
`wget https://dl.grafana.com/oss/release/grafana_6.5.1_arm64.deb`
`dpkg -i grafana_6.5.1_arm64.deb`
`systemctl enable grafana-server`
`systemctl start grafana-server`

As said it worked flawlessly for two weeks, now it won't start. How to find the point of error?

**Comment by aknuds1** [comment]
We need to find the originating error before we can come to any conclusion.

**Comment by aknuds1** [comment]
I find it strange that nothing is logged to /var/grafana/grafana.log before it crashes.

**Comment by aknuds1** [comment]
What's the output from the following command: `sudo systemctl start grafana-server`? Can you see any details on errors there?

**Comment by MarkoBursic** [comment]
I will post more details in following days. The server is at remote place, I have do personally visit and then see.

**Comment by MarkoBursic** [comment]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x788a3c]
I guess it is related to Golang. I have reinstalled the Grafana and now it works again, all dashboards and setting, users are preserved. Have no idea what this error means.

```
pi@smartpi:~$` sudo journalctl -u grafana-server
-- Logs begin at Wed 2019-12-18 07:43:52 CET, end at Wed 2019-12-18 10:01:06 CET. --
Dec 18 10:00:51 smartpi systemd[1]: Started Grafana instance.
Dec 18 10:00:51 smartpi grafana-server[2202]: t=2019-12-18T10:00:51+0100 lvl=info msg="Starting Grafana" logger=server version
Dec 18 10:00:51 smartpi grafana-server[2202]: panic: runtime error: invalid memory address or nil pointer dereference
Dec 18 10:00:51 smartpi grafana-server[2202]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x788a3c]
Dec 18 10:00:51 smartpi grafana-server[2202]: goroutine 1 [running]:
Dec 18 10:00:51 smartpi grafana-server[2202]: github.com/inconshreveable/log15.FilterHandler.func1(0x400054aea0, 0x0, 0x0)
Dec 18 10:00:51 smartpi grafana-server[2202]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2202]: github.com/inconshreveable/log15.funcHandler.Log(0x40006226a0, 0x400054aea0, 0x0
Dec 18 10:00:51 smartpi grafana-server[2202]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2202]: github.com/inconshreveable/log15.MultiHandler.func1(0x400054aea0, 0xc0, 0xc8)
Dec 18 10:00:51 smartpi grafana-server[2202]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2202]: github.com/inconshreveable/log15.funcHandler.Log(0x40006226e0, 0x400054aea0, 0x0
Dec 18 10:00:51 smartpi grafana-server[2202]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2202]: github.com/inconshreveable/log15.(*swapHandler).Log(0x4000068f40, 0x400054aea0, 
Dec 18 10:00:51 smartpi grafana-server[2202]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2202]: github.com/inconshreveable/log15.(*swapHandler).Log(0x40000fdbf0, 0x400054aea0, 
Dec 18 10:00:51 smartpi grafana-server[2202]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2202]: github.com/inconshreveable/log15.(*logger).write(0x40001cb320, 0x40005dbe10, 0x1
Dec 18 10:00:51 smartpi grafana-server[2202]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2202]: github.com/inconshreveable/log15.(*logger).Info(0x40001cb320, 0x40005dbe10, 0x10
Dec 18 10:00:51 smartpi grafana-server[2202]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2202]: main.(*Server).loadConfiguration(0x40004b8b40)
Dec 18 10:00:51 smartpi grafana-server[2202]:         /go/src/github.com/grafana/grafana/pkg/cmd/grafana-server/server.go:259 
Dec 18 10:00:51 smartpi grafana-server[2202]: main.(*Server).Run(0x40004b8b40, 0x0, 0x0)
Dec 18 10:00:51 smartpi grafana-server[2202]:         /go/src/github.com/grafana/grafana/pkg/cmd/grafana-server/server.go:85 +
Dec 18 10:00:51 smartpi grafana-server[2202]: main.main()
Dec 18 10:00:51 smartpi grafana-server[2202]:         /go/src/github.com/grafana/grafana/pkg/cmd/grafana-server/main.go:117 +0
Dec 18 10:00:51 smartpi systemd[1]: grafana-server.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Dec 18 10:00:51 smartpi systemd[1]: grafana-server.service: Failed with result 'exit-code'.
Dec 18 10:00:51 smartpi systemd[1]: grafana-server.service: Service RestartSec=100ms expired, scheduling restart.
Dec 18 10:00:51 smartpi systemd[1]: grafana-server.service: Scheduled restart job, restart counter is at 1.
Dec 18 10:00:51 smartpi systemd[1]: Stopped Grafana instance.
Dec 18 10:00:51 smartpi systemd[1]: Started Grafana instance.
Dec 18 10:00:51 smartpi grafana-server[2214]: t=2019-12-18T10:00:51+0100 lvl=info msg="Starting Grafana" logger=server version
Dec 18 10:00:51 smartpi grafana-server[2214]: panic: runtime error: invalid memory address or nil pointer dereference
Dec 18 10:00:51 smartpi grafana-server[2214]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x788a3c]
Dec 18 10:00:51 smartpi grafana-server[2214]: goroutine 1 [running]:
Dec 18 10:00:51 smartpi grafana-server[2214]: github.com/inconshreveable/log15.FilterHandler.func1(0x4000694410, 0x0, 0x0)
Dec 18 10:00:51 smartpi grafana-server[2214]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2214]: github.com/inconshreveable/log15.funcHandler.Log(0x40001a2380, 0x4000694410, 0x0
Dec 18 10:00:51 smartpi grafana-server[2214]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2214]: github.com/inconshreveable/log15.MultiHandler.func1(0x4000694410, 0xc0, 0xc8)
Dec 18 10:00:51 smartpi grafana-server[2214]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2214]: github.com/inconshreveable/log15.funcHandler.Log(0x40001a23c0, 0x4000694410, 0x4
Dec 18 10:00:51 smartpi grafana-server[2214]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2214]: github.com/inconshreveable/log15.(*swapHandler).Log(0x40000683b0, 0x4000694410, 
Dec 18 10:00:51 smartpi grafana-server[2214]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2214]: github.com/inconshreveable/log15.(*swapHandler).Log(0x4000471b80, 0x4000694410, 
Dec 18 10:00:51 smartpi grafana-server[2214]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2214]: github.com/inconshreveable/log15.(*logger).write(0x40004a0920, 0x40000ba490, 0x1
Dec 18 10:00:51 smartpi grafana-server[2214]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2214]: github.com/inconshreveable/log15.(*logger).Info(0x40004a0920, 0x40000ba490, 0x10
Dec 18 10:00:51 smartpi grafana-server[2214]:         /go/src/github.com/grafana/grafana/vendor/github.com/inconshreveable/log
Dec 18 10:00:51 smartpi grafana-server[2214]: main.(*Server).loadConfiguration(0x40001cd220)
Dec 18 10:00:51 smartpi grafana-server[2214]:         /go/src/github.com/grafana/grafana/pkg/cmd/grafana-server/server.go:259 
Dec 18 10:00:51 smartpi grafana-server[2214]: main.(*Server).Run(0x40001cd220, 0x0, 0x0)
Dec 18 10:00:51 smartpi grafana-server[2214]:         /go/src/github.com/grafana/grafana/pkg/cmd/grafana-server/server.go:85 +
Dec 18 10:00:51 smartpi grafana-server[2214]: main.main()

```

**Comment by aknuds1** [comment]
Thanks for getting the logs! This is a memory access error in Grafana, I will investigate if it might be a bug.

**Comment by aknuds1** [comment]
@MarkoBursic It looks to me as if your logs are truncated at 127 columns. Could you please obtain the full, un-truncated, logs? It's difficult to pinpoint the exact issue without full logs.

**Comment by aknuds1** [comment]
You could try getting the complete logs like this for example: `sudo journalctl -u grafana-server > logs.txt`.

**Comment by MarkoBursic** [comment]
I will try next time it breaks and report the full log.

**Comment by aknuds1** [comment]
Thanks, that would make it easier to figure out.

**Comment by MarkoBursic** [comment]
This is the result of `sudo journalctl -u grafana-server > logs.txt`

```
-- Logs begin at Tue 2020-01-07 15:08:07 CET, end at Tue 2020-01-07 17:22:29 CET. --
Jan 07 17:22:19 smartpi systemd[1]: Started Grafana instance.
Jan 07 17:22:19 smartpi grafana-server[2854]: Failed to start grafana. error: failed to initialize file handler: open /var/log/grafana/grafana.log: permission denied
Jan 07 17:22:19 smartpi systemd[1]: grafana-server.service: Main process exited, code=exited, status=1/FAILURE
Jan 07 17:22:19 smartpi systemd[1]: grafana-server.service: Failed with result 'exit-code'.
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Service RestartSec=100ms expired, scheduling restart.
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Scheduled restart job, restart counter is at 1.
Jan 07 17:22:20 smartpi systemd[1]: Stopped Grafana instance.
Jan 07 17:22:20 smartpi systemd[1]: Started Grafana instance.
Jan 07 17:22:20 smartpi grafana-server[2864]: Failed to start grafana. error: failed to initialize file handler: open /var/log/grafana/grafana.log: permission denied
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Main process exited, code=exited, status=1/FAILURE
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Failed with result 'exit-code'.
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Service RestartSec=100ms expired, scheduling restart.
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Scheduled restart job, restart counter is at 2.
Jan 07 17:22:20 smartpi systemd[1]: Stopped Grafana instance.
Jan 07 17:22:20 smartpi systemd[1]: Started Grafana instance.
Jan 07 17:22:20 smartpi grafana-server[2872]: Failed to start grafana. error: failed to initialize file handler: open /var/log/grafana/grafana.log: permission denied
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Main process exited, code=exited, status=1/FAILURE
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Failed with result 'exit-code'.
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Service RestartSec=100ms expired, scheduling restart.
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Scheduled restart job, restart counter is at 3.
Jan 07 17:22:20 smartpi systemd[1]: Stopped Grafana instance.
Jan 07 17:22:20 smartpi systemd[1]: Started Grafana instance.
Jan 07 17:22:20 smartpi grafana-server[2880]: Failed to start grafana. error: failed to initialize file handler: open /var/log/grafana/grafana.log: permission denied
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Main process exited, code=exited, status=1/FAILURE
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Failed with result 'exit-code'.
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Service RestartSec=100ms expired, scheduling restart.
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Scheduled restart job, restart counter is at 4.
Jan 07 17:22:20 smartpi systemd[1]: Stopped Grafana instance.
Jan 07 17:22:20 smartpi systemd[1]: Started Grafana instance.
Jan 07 17:22:20 smartpi grafana-server[2890]: Failed to start grafana. error: failed to initialize file handler: open /var/log/grafana/grafana.log: permission denied
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Main process exited, code=exited, status=1/FAILURE
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Failed with result 'exit-code'.
Jan 07 17:22:21 smartpi systemd[1]: grafana-server.service: Service RestartSec=100ms expired, scheduling restart.
Jan 07 17:22:21 smartpi systemd[1]: grafana-server.service: Scheduled restart job, restart counter is at 5.
Jan 07 17:22:21 smartpi systemd[1]: Stopped Grafana instance.
Jan 07 17:22:21 smartpi systemd[1]: grafana-server.service: Start request repeated too quickly.
Jan 07 17:22:21 smartpi systemd[1]: grafana-server.service: Failed with result 'exit-code'.
Jan 07 17:22:21 smartpi systemd[1]: Failed to start Grafana instance.
```

**Comment by aknuds1** [comment]
Do you know why Grafana lacks the permissions to open /var/log/grafana/grafana.log?

```
Jan 07 17:22:19 smartpi grafana-server[2854]: Failed to start grafana. error: failed to initialize file handler: open /var/log/grafana/grafana.log: permission denied
Jan 07 17:22:19 smartpi systemd[1]: grafana-server.service: Main process exited, code=exited, status=1/FAILURE
Jan 07 17:22:19 smartpi systemd[1]: grafana-server.service: Failed with result 'exit-code'.
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Service RestartSec=100ms expired, scheduling restart.
Jan 07 17:22:20 smartpi systemd[1]: grafana-server.service: Scheduled restart job, restart counter is at 1.
Jan 07 17:22:20 smartpi systemd[1]: Stopped Grafana instance.
```

**Comment by MarkoBursic** [comment]
I have no idea how to change file permissions. What do I need to revert the permission back? I did install `grafana_6.5.2_arm64.deb` over `grafana_6.5.1_arm64.deb` and it was working for two weeks. I thought it stopped this time for same reason, but it seems that there is a new issue.

**Comment by aknuds1** [comment]
Try `ls -l /var/log/grafana/grafana.log` to see what the permissions are. I would also ask such support questions in the [community forums](https://community.grafana.com/).

**Comment by shubh-07** [comment]
Hi, I am facing the same issue, 
`panic: runtime error: invalid memory address or nil pointer dereference
Apr 10 08:56:38 server1 grafana-server[27671]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x83e542]
Apr 10 08:56:38 server1 grafana-server[27671]: goroutine 1 [running]:
Apr 10 08:56:38 server1 grafana-server[27671]: github.com/grafana/grafana/pkg/setting.(*Cfg).loadConfiguration(0xc0004ae800, 0xc0004f1c78,
Apr 10 08:56:38 server1 grafana-server[27671]: /go/src/github.com/grafana/grafana/pkg/setting/setting.go:499 +0x5d2
Apr 10 08:56:38 server1 grafana-server[27671]: github.com/grafana/grafana/pkg/setting.(*Cfg).Load(0xc0004ae800, 0xc0004f1c78, 0x1aa80d8, 0
Apr 10 08:56:38 server1 grafana-server[27671]: /go/src/github.com/grafana/grafana/pkg/setting/setting.go:581 +0x61
Apr 10 08:56:38 server1 grafana-server[27671]: main.(*GrafanaServerImpl).loadConfiguration(0xc00052d570)
Apr 10 08:56:38 server1 grafana-server[27671]: /go/src/github.com/grafana/grafana/pkg/cmd/grafana-server/server.go:172 +0xe0
Apr 10 08:56:38 server1 grafana-server[27671]: main.(*GrafanaServerImpl).Run(0xc00052d570, 0x184b1e8, 0xc00052d570)
Apr 10 08:56:38 server1 grafana-server[27671]: /go/src/github.com/grafana/grafana/pkg/cmd/grafana-server/server.go:72 +0x43
Apr 10 08:56:38 server1 grafana-server[27671]: main.main()
Apr 10 08:56:38 server1 grafana-server[27671]: /go/src/github.com/grafana/grafana/pkg/cmd/grafana-server/main.go:95 +0x316
Apr 10 08:56:38 server1 audispd[1313]: node=server1 type=SERVICE_START msg=audit(1586526998.449:16821): pid=1 uid=0
Apr 10 08:56:38 server1 systemd[1]: grafana-server.service: main process exited, code=exited, status=2/INVALIDARGUMENT
Apr 10 08:56:38 server1 systemd[1]: Failed to start Grafana instance.`

Have your issue resolved??
MarkoBursic

**Comment by MarkoBursic** [comment]
@shubh-07 Not yet. Did you get the logs as @aknuds1 describes: `sudo journalctl -u grafana-server > logs.txt` . My box is located at the remote site, so the guy opened the log with `sudo nano` and then stored it with user privileges, that's why it got the lack of permission issue. The grafana was re-installed and worked, but I have been told it has stopped again. I am waiting to get VPN connection to the remote site, so I will have the possibility to extract logs by myself.

What OS you are running?

**Comment by shubh-07** [comment]
My issue is resolved!!!
@MarkoBursic Hi. My server is running on Linux (RHEL) OS. My systemd grafana service was working fine for about an year but recently there were some cleanups and sanity tests happened in the server due to which the user grafana lost rights. 
So I resolved this issue by changing the user and group in the grafana-server.service file to root and after daemon-reload the service works fine retaining all the dashboards plugins and data. 
So I can suggest two things two you. First is you can check the user permissions of grafana or either you can change the user in the service file or else just try to update the grafana on your system.

DO let me know if this resolves your issue or your errors prevail!!

**Comment by MarkoBursic** [comment]
Dear @shubh-07 

I get problems with: open /var/log/grafana/grafana.log: permission denied
Can you provide a step by step solution. It seems, that OS somehow overwrites permissions of specified directory to root. 
How can I change the user and group to root?

**Comment by shubh-07** [comment]
Hi @MarkoBursic 
Open the systemmd file for grafana. In my case, it is at the location /usr/lib/systemd/system/grafana-server.service . 
![image](https://user-images.githubusercontent.com/36189814/79984313-9d8ee480-84c6-11ea-87b9-3394a4360056.png)
Here you will find the user and group for the grafana service. Update this to root. 
Earlier in my case also the user was Grafana, and later i find out that grafana user didn't have the permissions due to this I was facing this error -> status=2/INVALIDARGUMENT. 
After changing this and then restarting the service the error resolved.

**Comment by MarkoBursic** [comment]
@shubh-07 Thank you. I have edited, now. We'll see next days if it stops again. As for now it started immediately without having to re-install. I will report the result and close this issue in few days.

**Comment by MarkoBursic** [comment]
It works now. Thank you for your support.

**Comment by upinlearntocode** [comment]
> Hi @MarkoBursic
> Open the systemmd file for grafana. In my case, it is at the location /usr/lib/systemd/system/grafana-server.service .
> ![image](https://user-images.githubusercontent.com/36189814/79984313-9d8ee480-84c6-11ea-87b9-3394a4360056.png)
> Here you will find the user and group for the grafana service. Update this to root.
> Earlier in my case also the user was Grafana, and later i find out that grafana user didn't have the permissions due to this I was facing this error -> status=2/INVALIDARGUMENT.
> After changing this and then restarting the service the error resolved.

Thank you sir, I have the same problem and it's worked.
